src/sql/workbench/services/query/common/queryRunner.ts

```diff
@@ -483,8 +483,8 @@ export default class QueryRunner extends Disposable {
 	 * @param resultId The result id of the result to copy from
 	 * @param includeHeaders [Optional]: Should column headers be included in the copy selection
 	 */
-	async copyResults(selections: Slick.Range[], batchId: number, resultId: number, includeHeaders?: boolean): Promise<void> {
-		await this.queryManagementService.copyResults({
+	async copyResults(selections: Slick.Range[], batchId: number, resultId: number, includeHeaders?: boolean): Promise<azdata.CopyResultsRequestResult> {
+		return await this.queryManagementService.copyResults({
```

Comment:
the await here shouldn't be needed, since whatever is calling this function will await the returned promise

Reply:
Does that mean I can remove async from the function declaration too?

Reply:
I think it's still needed. You can try removing it and seeing if there's an error. Here's an example of another place that directly returns the promise https://github.com/microsoft/azuredatastudio/blob/98c17b85e789d59488b778f9a7432773231fb9f8/extensions/sql-database-projects/src/common/utils.ts#L338-L341